src/client/pages/Section/hooks/useGetTableData/useDependencies.ts

```diff
@@ -71,10 +71,9 @@ export const useDependencies = (props: Props): Returned => {
 
         internal.tableNames.add(tableName)
         staticDependencies[tableName]?.forEach((t) => internal.tableNames.add(t))
-        if (withOdp) {
-          internal.tableNames.add(TableNames.originalDataPointValue)
-          internal.tableWithOdp = tableName
-        }
+
+        if (forestCharacteristicsUseOdp) internal.tableNames.add(TableNames.originalDataPointValue)
```

Comment:
Maybe, I am trying to understand the logic:

```
        const withOdp =
          tableName === TableNames.extentOfForest ||
          (tableName === TableNames.forestCharacteristics && forestCharacteristicsUseOdp)

        if (withOdp) {
           internal.tableWithOdp = tableName
           internal.tableNames.add(TableNames.originalDataPointValue)
         }
```

This is obviously not working when looking for tables outside of group `[1a, 1b]`.

My reasoning for the above correction (correct me if I am wrong):

```
        if (forestCharacteristicsUseOdp) internal.tableNames.add(TableNames.originalDataPointValue)
        if (withOdp) internal.tableWithOdp = tableName
```

Here we have divided the `if` clause in 2 parts:

First part:

If we have enabled ODPs in table `1b`, we will include the `originalDataPointValue` table in the `internal` deps.
This affects the tables fetched: it will include the ODP values and the validations should work as expected.

Second part:
We leave `tableWithOdp` inside `withOdp` clause:
This is used here:

```
        if (tableWithOdp && canEdit) {
          dispatch(DataActions.getNodeValuesEstimations({ ...propsFetch, sectionName, tableName: tableWithOdp }))
          dispatch(DataActions.getODPLastUpdatedTimestamp({ ...propsFetch, sectionName }))
        }
```

It seems we only use this information for fetching estimations and last update of any ODP

It seems tableWithOdp can be one of group `[extentOFForest, forestCharacteristics]` if I didn't misinterpret something


Let me know

@minotogna

Reply:
3a

Reply:
I will investigate deeper, but are we sure we want to set this variable in that case?

```
        if (withOdp) internal.tableWithOdp = tableName
```

Reply:
updated;

if odps are enabled and dependencies contain either `extentOfForest` or `forestCharacteristics` we add `originalDataPointValue`